Dz_4_5/src/bohdan/homework/com/Arena/Arena.java

```diff
@@ -8,69 +8,77 @@
-            if(winner[0] != null) return winner[0];
-            preRoundActions(arenaFighterTwo, arenaFighterOne, (winner1) -> winner[0] = winner1);
-            if(winner[0] != null) return winner[0];
-
-            int takeDamageOne = arenaFighterTwo.takeDamage(arenaFighterOne.getDamage());
-            int takeDamageTwo = arenaFighterOne.takeDamage(arenaFighterTwo.getDamage());
+    public ArenaFighter fight(ArenaFighter arenaFighterOne, ArenaFighter arenaFighterTwo) {
 
-            postRoundActions(arenaFighterOne,arenaFighterTwo,takeDamageTwo );
-            postRoundActions(arenaFighterTwo,arenaFighterOne, takeDamageOne);
+        int round = 1;
 
-            if (arenaFighterOne.isAlive() && arenaFighterTwo.isAlive())
-                round++;
 
-            else if (arenaFighterOne.isAlive()) {
-                System.out.println("Rounds passed = " + round);
+        final ArenaFighter[] winner = {null};
+        try {
+            do {
+                preRoundActions(arenaFighterOne, arenaFighterTwo, (winner1) -> winner[0] = winner1);
+                if (winner[0] != null) break;
+                preRoundActions(arenaFighterTwo, arenaFighterOne, (winner1) -> winner[0] = winner1);
+                if (winner[0] != null) break;
+
+                int takeDamageOneFighter = arenaFighterTwo.takeDamage(arenaFighterOne.getDamage()); //deals damage
+                int takeDamageTwoFighter = arenaFighterOne.takeDamage(arenaFighterTwo.getDamage()); //deals damage
+
+                postRoundActions(arenaFighterOne, arenaFighterTwo, takeDamageTwoFighter);
+                postRoundActions(arenaFighterTwo, arenaFighterOne, takeDamageOneFighter);
+
+
+                if (arenaFighterOne.isAlive() && arenaFighterTwo.isAlive())
+                    round++;
+
+                else if (arenaFighterOne.isAlive()) {
+                    System.out.println("Rounds passed = " + round);
+                    System.out.println("Winner = " + arenaFighterOne.getName() + "; Left health point = " + arenaFighterOne.getHealth());
+                    arenaFighterOne.setRegenHealth(arenaFighterOne.getFullHP());
+                    arenaFighterOne.setReturnDamage(arenaFighterOne.getFullDamage());
+                    return arenaFighterOne;
+                } else if (arenaFighterTwo.isAlive()) {
+                    System.out.println("Rounds passed = " + round);
+                    System.out.println("Winner = " + arenaFighterTwo.getName() + "; Left health point = " + arenaFighterTwo.getHealth());
+                    arenaFighterTwo.setRegenHealth(arenaFighterTwo.getFullHP());
+                    arenaFighterTwo.setReturnDamage(arenaFighterTwo.getFullDamage());
+                    return arenaFighterTwo;
+                }
+
+            } while (round < 11);
+        } catch (NullPointerException error) {
+            System.out.println("There is no winner");
+        }
+
+            if (arenaFighterOne.getHealth() > arenaFighterTwo.getHealth()) {
```

Comment:
Maybe for the winner, will be useful to use "winner[0]" varible to make code clean?